plasma_framework/test/endToEndTests/QuasarExit.e2e.test.js

```diff
@@ -0,0 +1,1082 @@
+            this.merkleProofForDepositTx = this.merkleTreeForDepositTx.getInclusionProof(this.depositTx);
+
+            return this.ethVault.deposit(this.depositTx, { from: alice, value: DEPOSIT_VALUE });
+        };
+
+        const aliceTransferEth = async (receiver, transferAmount) => {
+            const tranferTxBlockNum = (await this.framework.nextChildBlock()).toNumber();
+            this.transferUtxoPos = buildUtxoPos(tranferTxBlockNum, 0, 0);
+
+            const receiverOutput = new PaymentTransactionOutput(
+                OUTPUT_TYPE_PAYMENT,
+                transferAmount,
+                receiver,
+                ETH,
+            );
+            const outputAlice = new PaymentTransactionOutput(
+                OUTPUT_TYPE_PAYMENT, DEPOSIT_VALUE - transferAmount, alice, ETH,
+            );
+            if (transferAmount === DEPOSIT_VALUE) {
+                this.transferTxObject = new PaymentTransaction(1, [this.depositUtxoPos], [receiverOutput]);
+            } else {
+                this.transferTxObject = new PaymentTransaction(1, [this.depositUtxoPos], [receiverOutput, outputAlice]);
+            }
+            this.transferTx = web3.utils.bytesToHex(this.transferTxObject.rlpEncoded());
+            this.merkleTreeForTransferTx = new MerkleTree([this.transferTx]);
+            this.merkleProofForTransferTx = this.merkleTreeForTransferTx.getInclusionProof(this.transferTx);
+
+            await this.framework.submitBlock(this.merkleTreeForTransferTx.root, { from: authority });
+        };
+
+        describe('Given contracts deployed, exit game and ETH vault registered', () => {
+            before(setupContracts);
+
+            describe('And Quasar maintainer deploys and adds liquid funds to the Quasar', () => {
+                before(async () => {
+                    await setupQuasar();
+                    this.quasarCapacityBeforeAddingFunds = new BN(await this.quasar.tokenUsableCapacity(ETH));
+                    await this.quasar.addEthCapacity({ from: quasarMaintainer, value: QUASAR_LIQUID_FUNDS });
+                });
+
+                it('should increase the capacity of the Quasar', async () => {
+                    const quasarCapacityAfterAddingFunds = new BN(await this.quasar.tokenUsableCapacity(ETH));
+                    const quasarExpectedCapacity = this.quasarCapacityBeforeAddingFunds.addn(QUASAR_LIQUID_FUNDS);
+
+                    expect(quasarCapacityAfterAddingFunds).to.be.bignumber.equal(
+                        quasarExpectedCapacity,
+                    );
+                });
+
+                describe('When Alice deposited ETH to Vault', () => {
+                    before(async () => {
+                        await aliceDepositsETH();
+                    });
+
+                    describe('And then Alice tries to obtain a ticket from the Quasar using the output', () => {
+                        describe('If the output is from block over the safe-blocknum', () => {
```

Comment:
Exactly!
yes, the `updateSafeBlockLimit` is a bit cumbersome. One safe way we thought about was-
```
In the current design, there is a requirement for the Quasar owner to constantly keep updating the Safe Blocknum upper-bound. 
An alternative could be - Provide tickets for a UTXO if either one of these satisfy:

- UTXOs is older than (Latest_Block_timestamp - Buffer), ideally the buffer could take the value of the minimum validation period. or
- UTXO is from a safe block (determined by safe blocknum upper-bound) set by the Owner
```
Yes, i feel the same about your points too. since we are the only one running this for now
about 2) Yes this is a good idea, since we run our own watcher, we could run a service that could ping our watcher periodically to update the safe block limit